py/desisurvey/afternoonplan.py

```diff
@@ -47,108 +49,69 @@ def __init__(self, MJDstart, MJDend, surveycal, tilesubset=None):
         #   'EXPOSEFAC'; format = 'E'
         #   'PROGRAM'; format = '6A'
         #   'OBSCONDITIONS'; format = 'J'
-        #hdulist0 = pyfits.open(resource_filename('desimodel', 'data/footprint/desi-tiles.fits'))
-        hdulist0 = pyfits.open(os.path.join(os.environ['DESIMODEL'], 'data/footprint/desi-tiles.fits'))
-#        hdulist0 = pyfits.open('/Users/mlandriau/DESI/desihub/desimodel/py/desimodel/data/footprint/desi-tiles.fits')
-        tiledata0 = hdulist0[1].data
-        # This works because original table has index = tileID.
+        try:
+            tiles_path = os.path.join(
+                os.environ['DESIMODEL'], 'data', 'footprint', 'desi-tiles.fits')
+        except KeyError:
+            raise RuntimeError('DESIMODEL environment variable is not defined.')
+        tiles = astropy.table.Table.read(tiles_path, hdu=1)
+
+        # Only use tables with IN_DESI set.
+        keep = tiles['IN_DESI'] > 0
+
+        # Ignore EXTRA tiles.
+        keep[tiles['PROGRAM'] == 'EXTRA'] = False
```

Comment:
FYI, the "standard" but apparently not well known way to load `desi-tiles.fits` and trim to just tiles in the DESI footprint while discarding the PROGRAM='EXTRA' tiles:
```python
import desimodel.io
tiles = desimodel.io.load_tiles()
```
That returns an `astropy.io.fits.fitsrec.FITS_rec`; wrap with `astropy.table.Table()` if you'd prefer that.
Use keywords `onlydesi=False` and/or `extra=True` to get non-DESI and/or extra tiles.